### PR TITLE
tym: init at 3.3.0

### PR DESCRIPTION
**Pull request by wesleyjrz**
###### Description of changes

It's a lua-configurable terminal emulator base on VTE.
More information [here](https://github.com/endaaman/tym).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/tym/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchurl, pkg-config, gtk3, vte, lua5_3, pcre2 }:
+
+stdenv.mkDerivation rec {
+  pname = "tym";
+  version = "3.3.0";
+
+  src = fetchurl {
+    url = "https://github.com/ndaaman/${pname}/releases/download/${version}/${pname}-${version}.tar.gz";
```
Comment (applePrincess) [inline pkgs/applications/terminal-emulators/tym/default.nix:8]: ndaaman maight have changed their name to endaaman. if so update url.

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 176276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tym</li>
  </ul>
</details>

**Review by applePrincess — APPROVED** [review]

File: pkgs/applications/terminal-emulators/tym/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchurl, pkg-config, gtk3, vte, lua5_3, pcre2 }:
+
+stdenv.mkDerivation rec {
+  pname = "tym";
+  version = "3.3.0";
+
+  src = fetchurl {
```
Comment (onny) [inline pkgs/applications/terminal-emulators/tym/default.nix:7]: Is it possible to use `fetchFromGitHub` here instead?

Comment (wesleyjrz) [reply pkgs/applications/terminal-emulators/tym/default.nix:7]: I think it's possible, I tried it but I wasn't sure how to make it works, but now I think I can fix this. Should I send a new commit or just force amend and force push?

Comment (onny) [reply pkgs/applications/terminal-emulators/tym/default.nix:7]: amending is fine :)

Comment (wesleyjrz) [reply pkgs/applications/terminal-emulators/tym/default.nix:7]: Well, I think they don't have the make files in the repository, just in the releases, I don't know how to generate the necessary makefiles yet, could you help me?

Comment (applePrincess) [reply pkgs/applications/terminal-emulators/tym/default.nix:7]: It has `configure.ac`, so I think just adding `autoreconfHook` to `nativeBuildInputs` will solve

Comment (wesleyjrz) [reply pkgs/applications/terminal-emulators/tym/default.nix]: Done, thank you @applePrincess and @onny

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 176276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tym</li>
  </ul>
</details>

**Comment by onny** [comment]
One last small thing: The commit of tym contains changes to `maintainers/maintainer-list.nix`, please try to resolve this :)

File: pkgs/applications/terminal-emulators/tym/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, pkg-config, autoreconfHook, gtk3, vte, lua5_3, pcre2 }:
+
+stdenv.mkDerivation rec {
+  pname = "tym";
+  version = "3.3.0";
+
+  src = fetchFromGitHub {
+    owner = "endaaman";
+    repo = "${pname}";
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/tym/default.nix:10]:
```suggestion
    repo = "tym";
    rev = version;
```

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -14820,10 +14820,16 @@
...
+    github = "wesleyjrz";
+    githubId = 60184588;
+  };
   npatsakula = {
     email = "nikita.patsakula@gmail.com";
     name = "Patsakula Nikita";
     github = "npatsakula";
     githubId = 23001619;
-  };
+	};
```
Comment (applePrincess) [inline maintainers/maintainer-list.nix:14834]: This should be reverted.

**Comment by wesleyjrz** [comment]
Well, I did a soft reset and I think everything is right now.

**Comment by wesleyjrz** [comment]
> One last small thing: The commit of tym contains changes to `maintainers/maintainer-list.nix`, please try to resolve this :)

I did a soft reset, I think everything is right now.

**Pull request merged**